### PR TITLE
docs: add developer documentation to the lib/image-stream module

### PR DESCRIPTION
**Pull request by jviotti**
Signed-off-by: Juan Cruz Viotti <jviotti@openmailbox.org>

### CODE REVIEW

File: lib/image-stream/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+data, and useful metadata like the final image size.
+
+Archive Hooks
+-------------
+
+This module supports reading "archive images", which are defined by handlers
+(like `application/zip`). In order to avoid duplication on how to handle
+archives, archive support is implemented by "archive hooks".
+
+Archive hooks are CommonJS modules that expose two functions:
```
Comment (jhermsmeier) [inline lib/image-stream/README.md]: Could you also add how, and on what the archive hook modules would be instantiated here (i.e. are they called on the stream, or file descriptor, ...)? Or is that too much information, which should rather be somewhere else?

Comment (jviotti) [inline lib/image-stream/README.md]: Sounds like a good idea! I added an explanation for the handlers, and added proper function signatures to clarify the return type and arguments for the archive hook.

Notice that this module is currently coupled with the concept of a file name, which will be changed soon once we support streaming.

I have some designs for this in mind that I'll put on the relevant GH issue.

**Comment by jviotti** [comment]
@jhermsmeier Do you think this is good to merge now?

**Comment by jhermsmeier** [comment]
@jviotti yup, sorry for the delay!

**Pull request merged**